system/Database/BaseBuilder.php

```diff
@@ -895,15 +895,15 @@ public function orHavingNotIn(?string $key = null, $values = null, ?bool $escape
     protected function _whereIn(?string $key = null, $values = null, bool $not = false, string $type = 'AND ', ?bool $escape = null, string $clause = 'QBWhere')
     {
         if (empty($key) || ! is_string($key)) {
-            if (CI_DEBUG) {
+            if ($this->db->DBDebug) {
```

Comment:
I am not certain about these two. This isn't actually a database exception at all, it is a developer error. I'm actually surprised it was guarded behind `CI_DEBUG` to begin with - I think this should be thrown regardless. And invalid argument is a mistake no matter what the environment or context.

Reply:
Yes, this throws `InvalidArgumentException`, and in production do nothing.
Removing the `if` is a breaking change. Are you okay?

Reply:
If this exception is typed correctly then, yes: we should allow it. InvalidArgumentException is a type of LogicException which should never be suppressed. By definition (PHP.net):

> Exception that represents error in the program logic. This kind of exception should lead directly to a fix in your code.

If there is some conceivable execution path that could cause these two instances then they should be changed to RuntimeExceptions as allow suppression, but my impression looking at them is that we should just let them throw.

Reply:
Agreed. LogicException should not be suppressed.

Removed `if`.